### DISCUSSION TITLE
Compiling Mermaid in Latex

### DISCUSSION THREAD

**Discussion by acasanova99** [mermaid-js/mermaid#4766] [Ideas]
There is any way to generate latex code from a mermaid diagram? I mean a direct way of compiling it. It would be rather useful to have one. 

At the moment, I just generate an svg and add it as a picture.

**Comment by aloisklink** [comment]
~Not that I'm aware of.~ _Edit_: @Kochise's https://github.com/Kochise/latex-mermaid looks pretty good to me!

~It should in theory be possible to modify a package like [gnuplottex](https://ctan.org/tex-archive/macros/latex/contrib/gnuplottex) so you could just write:~

```latex
\begin{mermaid}
  %% your mermaid diagram here
\end{mermaid}
```

~Then your LaTeX package could take care of calling [mermaid-cli](https://github.com/mermaid-js/mermaid-cli) using `\immediate\write18{mermaid-cli\space \inputfolder/\figname.mmd \outputfolder/\figname.pdf}` to create a PDF, then import that PDF into your file.~

But, to be honest, if you're doing any complicated LaTeX project, you're probably using [`latexmk`](https://ctan.org/pkg/latexmk) with a custom `Makefile`, so in that case, it's probably just easier to add a rule to your `Makefile` like:

```Make
%.pdf: %.mermaid
    npx -p @mermaid-js/mermaid-cli mmdc --output $@ --input $<
```

Then, you can just run `\includegraphics{myfile.pdf}` files in your LaTeX file, and if they don't exist, if you call `latexmk` with the `-use-make` option, it should automatically generate `myfile.pdf` from `myfile.mermaid`.

**Comment by Kochise** [comment]
https://github.com/Kochise?tab=repositories&q=mermaid
